### PR TITLE
Making sure that the ACNV modeled segment cannot have a segment mean …

### LINKED ISSUE #399: Cannot use caller on GATK ACNV seg files (in GATK CNV format) since segment mean can get pegged to zero.
Need to have a minimum value (eps) when converting from log.
```
java.lang.IllegalArgumentException: Segment Mean must be finite. 
       at org.broadinstitute.hellbender.utils.param.ParamUtils.isFinite(ParamUtils.java:191) 
       at org.broadinstitute.hellbender.tools.exome.ModeledSegment.<init>(ModeledSegment.java:22)
 ....snip...
```

So I look in the converted SM-74P56-sim-final.cnv.seg file (zero for segment mean):
```
/dsde/working/lichtens/acnv/out_case_purity_series_pd250/acnv/SM-74P56  21      36088368        36207150        6       0
```

So I see the following in the regular ACNV output:
```
lichtens@gsa5 ~$ egrep 36088368 /dsde/working/lichtens/acnv/out_case_purity_series_pd250/acnv/SM-74P56-sim-final.seg
/dsde/working/lichtens/acnv/out_case_purity_series_pd250/acnv/SM-74P56  21      36088368        36207150        6       0       -1770.3685      -4292.2942      -539.8601    NaN     NaN     NaN
```

### PR DESCRIPTION
**Pull request by LeeTL1220**
…that is zero in CR space.

Closes #399

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/exome/acnvconversion/ACNVModeledSegmentConversionUtils.java [hunk 1]
```diff
@@ -27,9 +27,23 @@ private ACNVModeledSegmentConversionUtils() {}
...
+
+        final TargetCollection<TargetCoverage> targets = genome.getTargets();
+        return convertACNVModeledSegmentToModeledSegment(acnvModeledSegment, targets);
+    }
+
+    @VisibleForTesting
+    static ModeledSegment convertACNVModeledSegmentToModeledSegment(ACNVModeledSegment acnvModeledSegment, TargetCollection<TargetCoverage> targets) {
+
+        // Make sure that we do not let segment mean become zero
+        double updatedCenter = acnvModeledSegment.getSegmentMeanPosteriorSummary().center();
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/acnvconversion/ACNVModeledSegmentConversionUtils.java]: If I understand correctly, this is not in log space yet. . .

File: src/main/java/org/broadinstitute/hellbender/tools/exome/acnvconversion/ACNVModeledSegmentConversionUtils.java [hunk 2]
```diff
@@ -27,9 +27,23 @@ private ACNVModeledSegmentConversionUtils() {}
...
+        if (Math.pow(2, updatedCenter) <= 0) {
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/acnvconversion/ACNVModeledSegmentConversionUtils.java]: . . . and it gets converted to log space (log2 of the `EPSILON` threshold) only if it enters the body of this `if` statement. . .

File: src/main/java/org/broadinstitute/hellbender/tools/exome/acnvconversion/ACNVModeledSegmentConversionUtils.java [hunk 3]
```diff
@@ -27,9 +27,23 @@ private ACNVModeledSegmentConversionUtils() {}
...
+            updatedCenter = ParamUtils.log2(TangentNormalizer.EPSILON);
+        }
+
         return new ModeledSegment(acnvModeledSegment.getInterval(), ModeledSegment.NO_CALL,
-                genome.getTargets().targetCount(acnvModeledSegment.getInterval()),
-                acnvModeledSegment.getSegmentMeanPosteriorSummary().center());
+                targets.targetCount(acnvModeledSegment.getInterval()), updatedCenter);
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/acnvconversion/ACNVModeledSegmentConversionUtils.java]: . . . so it seems that sometimes this constructor gets a log2 copy ratio and sometimes not.  Am I getting this wrong?

Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/acnvconversion/ACNVModeledSegmentConversionUtils.java]: Never mind previous comments, I see that `ACNVModeledSegment` starts out in log space.

File: src/main/java/org/broadinstitute/hellbender/tools/exome/acnvconversion/ACNVModeledSegmentConversionUtils.java (hunk 1 above)
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/acnvconversion/ACNVModeledSegmentConversionUtils.java]: How about simply `updatedCenter = Math.max(acnvModeledSegment.getSegmentMeanPosteriorSummary().center(), ParamUtils.log2(TangentNormalizer.EPSILON))`

**Comment by davidbenjamin** [comment]
@LeeTL1220  One question, do with it as you will and merge.

**Comment by LeeTL1220** [comment]
@davidbenjamin Comments addressed...

**Pull request merged**